src/systems/ticker.js

```diff
@@ -0,0 +1,23 @@
+AFRAME.registerSystem("ticker", {
+  init() {
+    this.promises = [];
+    this.nextPromises = [];
+  },
+
+  tick() {
+    for (let i = 0; i < this.promises.length; i++) {
+      this.promises[i]();
+    }
+    this.promises.length = 0;
+    for (let i = 0; i < this.nextPromises.length; i++) {
+      this.promises.push(this.nextPromises[i]);
+    }
+    this.nextPromises.length = 0;
+  },
+
+  nextTick() {
+    return new Promise(resolve => {
+      this.nextPromises.push(resolve);
+    });
+  }
+});
```

Comment:
Am I missing something, or does `nextTick` wait for the next next tick?

1. Initialize `promises` and `nextPromises` to be empty.
2. `nextTick` is called. `nextPromises` contains a resolver for the returned promise.
3. A tick happens. `nextPromises[0]` is moved into `promises[0]`.
4. Another tick happens. `promises[0]` is called and the promise resolves.

That's counterintuitive to me.

Reply:
Yes it does wait for the next next aframe system tick, and the reason for that is because I expect most requests to come in from component updates/ticks, which come BEFORE this system's tick. However I realized this is the wrong approach. I was using this system's `tick` as the timer, which has a few problems:
- If I call nextFrame before aframe starts ticking the system (but after it exists), then the promise resolves before the first render.
- If I call nextFrame within a system, then the behavior is different depending on whether the system ticks before or after this one.
- If I call nextFrame from some async code, it's not clear whether it's going to wait 1 or 2 ticks.

Instead what I'll do is overwrite the scene's tick or the scene's render, because that's what calls the gl render which I need to ensure happens before these promises resolve.